diff --git a/torch/csrc/jit/frontend/ir_emitter.cpp b/torch/csrc/jit/frontend/ir_emitter.cpp
index 4c066c27c3..d03fc4a767 100644
--- a/torch/csrc/jit/frontend/ir_emitter.cpp
+++ b/torch/csrc/jit/frontend/ir_emitter.cpp
@@ -1678,7 +1678,7 @@ struct to_ir {
               << "Union type annotation `" << type_hint->repr_str()
               << "` can hold " << vector_repr.str() << ", but none of "
               << "those list types can hold the types of the given dict"
-              << " elements, which were unified to " << candidate->repr_str();
+              << " elements, which were unified to " << (*unified_value_type)->repr_str();
         } else {
           refined_type_hint = candidate;
         }
