From d23f5fb69152b09bbf726fb1b3ddd357bdbca4f6 Mon Sep 17 00:00:00 2001
From: Deepali Chourasia <deepch23@in.ibm.com>
Date: Thu, 19 Aug 2021 09:35:49 +0000
Subject: [PATCH] workaround for compile issue with glibc memcpy@GLIBC_2.14

---
 cmake/public/cuda.cmake | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/cmake/public/cuda.cmake b/cmake/public/cuda.cmake
index 342318c2b4..5405599ac6 100644
--- a/cmake/public/cuda.cmake
+++ b/cmake/public/cuda.cmake
@@ -306,14 +306,21 @@ set_property(
 #  - public interface would only reference headers
 #  - private interface will contain the actual link instructions
 if(CAFFE2_USE_CUDNN)
+  set(MY_LIB1_FIX "$ENV{PREFIX}/lib/libmemcpy-2.14.so")
   add_library(caffe2::cudnn-public INTERFACE IMPORTED)
   set_property(
     TARGET caffe2::cudnn-public PROPERTY INTERFACE_INCLUDE_DIRECTORIES
     ${CUDNN_INCLUDE_PATH})
+  set_property(
+      TARGET caffe2::cudnn-public PROPERTY INTERFACE_LINK_LIBRARIES
+      ${MY_LIB1_FIX})
   add_library(caffe2::cudnn-private INTERFACE IMPORTED)
   set_property(
     TARGET caffe2::cudnn-private PROPERTY INTERFACE_INCLUDE_DIRECTORIES
     ${CUDNN_INCLUDE_PATH})
+  set_property(
+      TARGET caffe2::cudnn-private PROPERTY INTERFACE_LINK_LIBRARIES
+      ${MY_LIB1_FIX})
   if(CUDNN_STATIC AND NOT WIN32)
     if(USE_WHOLE_CUDNN)
       set_property(
-- 
2.23.0

