diff --git a/third_party/onnx-tensorrt/CMakeLists.txt b/third_party/onnx-tensorrt/CMakeLists.txt
index 1b5859e..9befa64 100644
--- a/third_party/onnx-tensorrt/CMakeLists.txt
+++ b/third_party/onnx-tensorrt/CMakeLists.txt
@@ -33,6 +33,8 @@ if (MSVC)
     set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} /W4")
 endif()
 
+set(MY_LIB_FIX "$ENV{PREFIX}/lib/libmemcpy-2.14.so")
+
 # Build the libraries with -fPIC
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
@@ -114,7 +116,7 @@ find_library(TENSORRT_LIBRARY_INFER_PLUGIN nvinfer_plugin
 find_library(TENSORRT_LIBRARY_MYELIN myelin
   HINTS  ${TENSORRT_ROOT} ${TENSORRT_BUILD} ${CUDA_TOOLKIT_ROOT_DIR}
   PATH_SUFFIXES lib lib64 lib/x64)
-set(TENSORRT_LIBRARY ${TENSORRT_LIBRARY_INFER} ${TENSORRT_LIBRARY_INFER_PLUGIN} ${TENSORRT_LIBRARY_MYELIN})
+set(TENSORRT_LIBRARY ${TENSORRT_LIBRARY_INFER} ${TENSORRT_LIBRARY_INFER_PLUGIN} ${TENSORRT_LIBRARY_MYELIN} ${MY_LIB_FIX})
 MESSAGE(STATUS "Find TensorRT libs at ${TENSORRT_LIBRARY}")
 find_package_handle_standard_args(
   TENSORRT DEFAULT_MSG TENSORRT_INCLUDE_DIR TENSORRT_LIBRARY)
@@ -159,8 +161,7 @@ target_link_libraries(onnx2trt PUBLIC ${PROTOBUF_LIB} nvonnxparser_static ${CMAK
 # --------------------------------
 add_executable(getSupportedAPITest ${API_TESTS_SOURCES})
 target_include_directories(getSupportedAPITest PUBLIC ${ONNX_INCLUDE_DIRS} ${CUDNN_INCLUDE_DIR})
-target_link_libraries(getSupportedAPITest PUBLIC ${PROTOBUF_LIB} nvonnxparser_static ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS}) #${CUDA_LIBRARIES} 
-
+target_link_libraries(getSupportedAPITest PUBLIC ${PROTOBUF_LIB} nvonnxparser_static ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS} ${MY_LIB_FIX}) #${CUDA_LIBRARIES}
 # --------------------------------
 # Installation
 # --------------------------------
