diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index ca560288a4..44771e5756 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -110,6 +110,10 @@ if(USE_TBB)
     set(TBB_INCLUDE_DIR "${TBB_ROOT_DIR}/include")
 
     add_library(TBB::tbb ALIAS tbb)
+
+    install(DIRECTORY ${TBB_INCLUDE_DIR}
+            DESTINATION ${CMAKE_INSTALL_PREFIX}
+            FILES_MATCHING PATTERN "*.h")
   endif()
 endif()
 
diff --git a/setup.py b/setup.py
index 6d9ed53dc6..8cfc2d3547 100644
--- a/setup.py
+++ b/setup.py
@@ -1047,7 +1047,7 @@ if __name__ == '__main__':
                 'utils/model_dump/skeleton.html',
                 'utils/model_dump/code.js',
                 'utils/model_dump/*.mjs',
-            ],
+            ] + (['include/tbb/*.h','include/tbb/combat/*.h','include/tbb/internal/*.h','include/tbb/machine/*.h'] if os.environ.get('USE_TBB') == '1' else []),
             'caffe2': [
                 'python/serialized_test/data/operator_test/*.zip',
             ],
diff --git a/torch/CMakeLists.txt b/torch/CMakeLists.txt
index 743a5d4421..28e9026bbe 100644
--- a/torch/CMakeLists.txt
+++ b/torch/CMakeLists.txt
@@ -72,6 +72,10 @@ set(TORCH_PYTHON_INCLUDE_DIRECTORIES
     ${TORCH_ROOT}/third_party/onnx
     ${pybind11_INCLUDE_DIRS}
 
+if(USE_TBB)
+   ${TBB_INCLUDE_DIR}
+endif()
+
     ${TORCH_SRC_DIR}/csrc
     ${TORCH_SRC_DIR}/csrc/api/include
     ${TORCH_SRC_DIR}/lib
