diff --git a/torch/utils/cpp_extension.py b/torch/utils/cpp_extension.py
index 227740e..0e4e420 100644
--- a/torch/utils/cpp_extension.py
+++ b/torch/utils/cpp_extension.py
@@ -1606,7 +1606,8 @@ def _write_ninja_file_to_build_library(path,
             if not any(flag.startswith('-std=') for flag in cuda_flags):
                 cuda_flags.append('-std=c++14')
             if os.getenv("CC") is not None:
-                cuda_flags = ['-ccbin', os.getenv("CC")] + cuda_flags
+                if '-ccbin' not in cuda_flags:
+                    cuda_flags = ['-ccbin', os.getenv("CC")] + cuda_flags
     else:
         cuda_flags = None
 
