diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index d396daf5f9..fcc0868325 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1240,6 +1240,7 @@ if (CAFFE2_CMAKE_BUILDING_WITH_MAIN_REPO)
     set(CMAKE_CUDA_COMPILER ${CUDA_NVCC_EXECUTABLE})
     add_onnx_tensorrt_subdir()
     include_directories("${CMAKE_CURRENT_LIST_DIR}/../third_party/onnx-tensorrt")
+    include_directories("${CMAKE_SOURCE_DIR}/caffe2/core/nomnigraph/include/")
     caffe2_interface_library(nvonnxparser_static onnx_trt_library)
     list(APPEND Caffe2_DEPENDENCY_WHOLE_LINK_LIBS onnx_trt_library)
     set(CAFFE2_USE_TRT 1)
